### ISSUE TITLE
Remove AI scripting from the botlib

### ISSUE THREAD

**Issue by ETLegacyTeam** [etlegacy/etlegacy#3]
Author: @IR4T4 (IR4T4)
Date: 2012-03-02
Redmine Issue: 3, https://dev.etlegacy.com/issues/3

---

As far as I know the botlib isn't used by any ET mod. Please remove. 

Note: Keep in mind some 'botlib' traps are still required:
 BOTLIB_PC_LOAD_SOURCE = 579,
 BOTLIB_PC_FREE_SOURCE,
 BOTLIB_PC_READ_TOKEN,
 BOTLIB_PC_SOURCE_FILE_AND_LINE,
 BOTLIB_PC_UNREAD_TOKEN

Here is a list of used system traps:
```
//
 // system traps provided by the main engine
 //
 typedef enum {
 //============== general Quake services ==================
G_PRINT = 0,        // ( const char *string );
// print message on the local console

G_ERROR,        // ( const char *string );
// abort the game

G_MILLISECONDS, // ( void );
// get current time for profiling reasons
// this should NOT be used for any game related tasks,
// because it is not journaled

// console variable interaction
G_CVAR_REGISTER,    // ( vmCvar_t *vmCvar, const char *varName, const char *defaultValue, int flags );
G_CVAR_UPDATE,  // ( vmCvar_t *vmCvar );
G_CVAR_SET,     // ( const char *var_name, const char *value );
G_CVAR_VARIABLE_INTEGER_VALUE,  // ( const char *var_name );

G_CVAR_VARIABLE_STRING_BUFFER,  // ( const char *var_name, char *buffer, int bufsize );

G_CVAR_LATCHEDVARIABLESTRINGBUFFER,

G_ARGC,         // ( void );
// ClientCommand and ServerCommand parameter access

G_ARGV,         // ( int n, char *buffer, int bufferLength );

G_FS_FOPEN_FILE,    // ( const char *qpath, fileHandle_t *file, fsMode_t mode );
G_FS_READ,      // ( void *buffer, int len, fileHandle_t f );
G_FS_WRITE,     // ( const void *buffer, int len, fileHandle_t f );
G_FS_RENAME,
G_FS_FCLOSE_FILE,       // ( fileHandle_t f );

G_SEND_CONSOLE_COMMAND, // ( const char *text );
// add commands to the console as if they were typed in
// for map changing, etc


//=========== server specific functionality =============

G_LOCATE_GAME_DATA,     // ( gentity_t *gEnts, int numGEntities, int sizeofGEntity_t,
//                          playerState_t *clients, int sizeofGameClient );
// the game needs to let the server system know where and how big the gentities
// are, so it can look at them directly without going through an interface

G_DROP_CLIENT,      // ( int clientNum, const char *reason );
// kick a client off the server with a message

G_SEND_SERVER_COMMAND,  // ( int clientNum, const char *fmt, ... );
// reliably sends a command string to be interpreted by the given
// client.  If clientNum is -1, it will be sent to all clients

G_SET_CONFIGSTRING, // ( int num, const char *string );
// config strings hold all the index strings, and various other information
// that is reliably communicated to all clients
// All of the current configstrings are sent to clients when
// they connect, and changes are sent to all connected clients.
// All confgstrings are cleared at each level start.

G_GET_CONFIGSTRING, // ( int num, char *buffer, int bufferSize );

G_GET_USERINFO,     // ( int num, char *buffer, int bufferSize );
// userinfo strings are maintained by the server system, so they
// are persistant across level loads, while all other game visible
// data is completely reset

G_SET_USERINFO,     // ( int num, const char *buffer );

G_GET_SERVERINFO,   // ( char *buffer, int bufferSize );
// the serverinfo info string has all the cvars visible to server browsers

G_SET_BRUSH_MODEL,  // ( gentity_t *ent, const char *name );
// sets mins and maxs based on the brushmodel name

G_TRACE,    // ( trace_t *results, const vec3_t start, const vec3_t mins, const vec3_t maxs, const vec3_t end, int passEntityNum, int contentmask );
// collision detection against all linked entities

G_POINT_CONTENTS,   // ( const vec3_t point, int passEntityNum );
// point contents against all linked entities

G_IN_PVS,           // ( const vec3_t p1, const vec3_t p2 );

G_IN_PVS_IGNORE_PORTALS,    // ( const vec3_t p1, const vec3_t p2 );

G_ADJUST_AREA_PORTAL_STATE, // ( gentity_t *ent, qboolean open );

G_AREAS_CONNECTED,  // ( int area1, int area2 );

G_LINKENTITY,       // ( gentity_t *ent );
// an entity will never be sent to a client or used for collision
// if it is not passed to linkentity.  If the size, position, or
// solidity changes, it must be relinked.

G_UNLINKENTITY,     // ( gentity_t *ent );
// call before removing an interactive entity

G_ENTITIES_IN_BOX,  // ( const vec3_t mins, const vec3_t maxs, gentity_t **list, int maxcount );
// EntitiesInBox will return brush models based on their bounding box,
// so exact determination must still be done with EntityContact

G_ENTITY_CONTACT,   // ( const vec3_t mins, const vec3_t maxs, const gentity_t *ent );
// perform an exact check against inline brush models of non-square shape

    #ifdef OMNIBOTS
// IRATA: used by OmniBot
G_BOT_ALLOCATE_CLIENT = 36, // ( int clientNum );
    .#endif

G_GET_USERCMD = 38, // ( int clientNum, usercmd_t *cmd )

G_GET_ENTITY_TOKEN, // qboolean ( char *buffer, int bufferSize )
// Retrieves the next string token from the entity spawn text, returning
// false when all tokens have been parsed.
// This should only be done at GAME_INIT time.

G_FS_GETFILELIST,
G_DEBUG_POLYGON_CREATE,
G_DEBUG_POLYGON_DELETE,
G_REAL_TIME,
G_SNAPVECTOR,
    // MrE:

G_TRACECAPSULE, // ( trace_t *results, const vec3_t start, const vec3_t mins, const vec3_t maxs, const vec3_t end, int passEntityNum, int contentmask );
// collision detection using capsule against all linked entities

G_ENTITY_CONTACTCAPSULE,    // ( const vec3_t mins, const vec3_t maxs, const gentity_t *ent );
// perform an exact check against inline brush models of non-square shape
   // done.

G_GETTAG,

G_REGISTERTAG,
// Gordon: load a serverside tag

G_REGISTERSOUND,    // xkan, 10/28/2002 - register the sound
G_GET_SOUND_LENGTH, // xkan, 10/28/2002 - get the length of the sound

   #ifdef OMNIBOTS
// 200
BOTLIB_GET_CONSOLE_MESSAGE = 210, // ( int client, char *message, int size );
BOTLIB_USER_COMMAND = 211,           // ( int client, usercmd_t *ucmd );

// 400
BOTLIB_EA_COMMAND = 407,
    #endif

// 500
// files
BOTLIB_PC_LOAD_SOURCE = 579,
BOTLIB_PC_FREE_SOURCE,
BOTLIB_PC_READ_TOKEN,
BOTLIB_PC_SOURCE_FILE_AND_LINE,
BOTLIB_PC_UNREAD_TOKEN,
   #ifdef PUNKBUSTER
// PB_STAT_REPORT = 584,
   #endif
// zinx
G_SENDMESSAGE = 585,
G_MESSAGESTATUS,
// -zinx

} gameImport_t;

```

---

Related issue(s): #31

---

Related commit(s):
1e50d88359c65b6310532728013c10d53ea175fa - partially removed botlib, fixes 3
 8558c910cd80e21cba47014fa235614ca89d8d73 - Revert "partially removed botlib, fixes 3"This ..
 96f272aa418912678223063ac3ef0e121e13f99e - first stage of botlib code removal, refs 3
 994b3f0101c9c1eba83c44412a24c02e687d555d - game bot removal refs 3
 1ca1c5fd1503cc3755ab189a609ab3956ce50b8a - premake updade bot removal refs 3
 ffef98bc33ea789190936ba3bebe4c0bcb4cdf51 - botlib removal refs 3
 b4fec651735ce0f7aa7c629b45e06ec9986961f4 - botlib removal refs 3
 ee6e151f2371c9926e0b5d18645431c4e995794e - botlib fixes refs 3
 7dedf7dc4a88d0d5a73cdea9f2894265c9ac94fc - botlib fixes refs 3
 5d6768e446367bd125448be911562b00deea04fc - botlib issues refs 3
 89553560e82a7ddcb1f5d01d0935104a21e3fc74 - missing function added refs 3
 9090305597cf4f03dd4a8fc2b5235af9b053d172 - let's use Q_stristr refs 3
 2554250c0d883020bc45ed51b348e5fa06574668 - botlib issues refs 3
 f7ce4edd6166df51db6f425d857d3f4f543db3b2 - constructible functions refs 3
 aa6519cc4a78363809f44026afe47d30622162f2 - botlib ... refs 3
 dcf4f482d4949097ddd853ee037235b240a22cff - botlib ... refs 3
 b2011044e5bd891da25baa487ff2c2af94b34b5c - botlib ... refs 3
 10a0785f8f8b9c220dd06f5d0914a5db11092d84 - botlib removal, refs 3
 05d95c51a3c54069fa7080aa1c69f93b5b5083ba - botlib removal ditched obsolete files refs 3
 4926f6635c13d8f8403a552f4718e9ea1382d893 - botlib kicked out of BOTLIB_EA_COMMAND case refs ..
 e5ea88025f50be8349fa30a940db77aa9267a10c - botlib removal refs 3
 9f12761704315dab8fc4a28cb408f60d0d15749d - botlib shrinked refs 3
 f796c990329e399a21e57bd3ebf4f4d4538fec27 - botlib is no longer required for game refs 3

**Comment by ETLegacyTeam** [comment]
From: @IR4T4 (IR4T4)
Date: 2012-03-16T09:51:53Z

---

... and gameExport_t
```<code class="C">
//
// functions exported by the game subsystem
//
typedef enum {
	GAME_INIT = 0,	// ( int levelTime, int randomSeed, int restart );
	// init and shutdown will be called every single level
	// The game should call G_GET_ENTITY_TOKEN to parse through all the
	// entity configuration text and spawn gentities.

	GAME_SHUTDOWN,	// (void);

	GAME_CLIENT_CONNECT,	// ( int clientNum, qboolean firstTime, qboolean isBot );
	// return NULL if the client is allowed to connect, otherwise return
	// a text string with the reason for denial

	GAME_CLIENT_BEGIN,				// ( int clientNum );

	GAME_CLIENT_USERINFO_CHANGED,	// ( int clientNum );

	GAME_CLIENT_DISCONNECT,			// ( int clientNum );

	GAME_CLIENT_COMMAND,			// ( int clientNum );

	GAME_CLIENT_THINK,				// ( int clientNum );

	GAME_RUN_FRAME,					// ( int levelTime );

	GAME_CONSOLE_COMMAND,			// ( void );
	// ConsoleCommand will be called when a command has been issued
	// that is not recognized as a builtin function.
	// The game can issue trap_argc() / trap_argv() commands to get the command
	// and parameters.  Return qfalse if the game doesn't recognize it as a command.

	GAME_SNAPSHOT_CALLBACK = 10,			// ( int entityNum, int clientNum ); // return qfalse if you don't want it to be added

	// IRATA: these gameExport_t are not used anymore
	// BOTAI_START_FRAME,              // ( int time ); - NOT USED IN NQ - keep for compatibility

	// Ridah, Cast AI
	//BOT_VISIBLEFROMPOS, // - NOT USED IN NQ - keep for compatibility
	//BOT_CHECKATTACKATPOS,//- NOT USED IN NQ - keep for compatibility
	// done.

	// zinx
	GAME_MESSAGERECEIVED = 14,			// ( int cno, const char *buf, int buflen, int commandTime );
	// -zinx
} gameExport_t;
</code>
```

**Comment by ETLegacyTeam** [comment]
From: @JanSimek (Radegast)
Date: 2012-03-17T14:04:15Z

---

commit:1e50d883 "partially removed botlib, fixes #3" broke compatibility with mods. 

I give up.

**Comment by ETLegacyTeam** [comment]
From: @IR4T4 (IR4T4)
Date: 2012-04-25T10:24:01Z

---

see 

http://etlegacy.com/projects/etlegacy/repository/revisions/96f272aa418912678223063ac3ef0e121e13f99e

to remove functions
